simple_rpc/simple_rpc.py

```diff
@@ -44,6 +44,35 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def _open(self):
+        if self.is_open():
+            return
+        try:
+            self._connection.open()
+        except SerialException as error:
+            raise IOError(error.strerror.split(':')[0])
+
+    def _close(self):
+        if not self.is_open():
```

Comment:
Guard for double closing. This will happen when we work with a socket (which auto-closes).

Reply:
ok